packages/@aws-cdk/aws-servicecatalogappregistry/lib/application.ts

```diff
@@ -0,0 +1,90 @@
+import * as cdk from '@aws-cdk/core';
+import { CfnApplication } from './servicecatalogappregistry.generated';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from 'constructs';
+
+/**
+ * A Service Catalog AppRegistry Application.
+ */
+export interface IApplication extends cdk.IResource {
+  /**
+   * The ARN of the application.
+   * @attribute
+   */
+  readonly applicationArn: string;
+
+  /**
+   * The ID of the application.
+   * @attribute
+   */
+  readonly applicationId: string;
+}
+
+/**
+ * Properties for a Service Catalog AppRegistry Application
+ */
+export interface ApplicationProps {
+  /**
+   * Enforces a particular physical application name.
+   */
+  readonly applicationName: string;
+
+  /**
+   * Description for application.
+   * @default - No description provided
+   */
+  readonly description?: string;
+}
+
+abstract class ApplicationBase extends cdk.Resource implements IApplication {
+  public abstract readonly applicationArn: string;
+  public abstract readonly applicationId: string;
+}
+
+/**
+ * A Service Catalog AppRegistry Application.
+ */
+export class Application extends ApplicationBase implements IApplication {
+  /**
+   * Imports an Application construct that represents an external application.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param applicationArn the Amazon Resource Name of the existing AppRegistry Application
+   */
+  public static fromApplicationArn(scope: Construct, id: string, applicationArn: string): IApplication {
+    const applicationId = applicationArn.split('/').pop()!;
```

Comment:
Want to follow up here. The primary reason I didn't use the `parseArn` was because I do not get both of the forward slashes in the resourceName, it yields `applications/0aqmvxvgmry0ecc4mjhwypun6i`.  So then we would still need to do something like check if the `arnResource` starts with a `/` or same sort of thing with splitting.  Either way would still need some weirder code for finding the applicationId from the name.  If using the `parseArn` and still checking if we can split or if it starts with the forward slash works please let me know.

Reply:
Hmm, you're right. I'm pretty sure this is a bug in `parseArn()`. Nice. catch!

I'll submit a PR with the fix. For now, just use the existing logic, and don't worry about detecting whether the `resourceName` of the parsed ARN starts with a `/`, or not.

Reply:
okay thanks! And actually I myself had to go on a deep dive.  It appears because this is a new service that they adopted a newer ARN standard that hasn't become official yet.  Hence I wouldn't have expected the standard library to support this yet.

Reply:
Yep, that's almost certainly the problem here.

Reply:
Fix: https://github.com/aws/aws-cdk/pull/15166

Reply:
Do I need to wait for that fix to go live? (and then fix it here).  Does not matter to me, but if the case I would like to add another construct that is basically a copy of this to this PR in mean time then.

Reply:
I wanted to do it this way, but looks like that PR will need some time. In that case, I don't want to hold your PR on it.

Let's just proceed with what we have here now (don't add another construct to this PR). I'll review the PR again today.

Reply:
So, the fix in https://github.com/aws/aws-cdk/pull/15166 has been merged.

Given that, can we use it here? It should be as simple as replacing `String.split('/')` here with

```suggestion
    const arn = cdk.Stack.of(scope).splitArn(applicationArn, cdk.ArnFormat.SLASH_RESOURCE_SLASH_RESOURCE_NAME);
    const applicationId = arn.resourceName;
```